### DISCUSSION TITLE
Is there a way to invoke /register during signin-exp?

### DISCUSSION THREAD

**Discussion by 0zul** [logto-io/logto#4583] [Q&A]
I have noticed that auth0 has a param called `screen_hint` if when set to signup it redirects to account creation page. I'd like to know if logto has that feature. If not it would be a really nice way to add that since it will save so much time.

**Comment by darcyYe** [comment]
Are you suggesting controlling whether to guide the user towards the sign-in flow or the sign-up flow through parameters? Logto has different routes for end users' sign-in/sign-up flows respectively.

**Reply by 0zul** [reply]
Yes exactly, for example, I want to send a user to sign up rather than sign in if they're a new user. I use next-auth as it's simple to use and I can easily manage sessions in it.

**Comment by mrfrase3** [comment]
For those that want to save the source code deep-dive that I just did:

when calling /oidc/auth you can add `interaction_mode=signUp` or `interaction_mode=signIn` to indicate whether to start the user on /register or /sign-in respectively

if using the SDK, it is the second parameter to the signIn method:
https://github.com/logto-io/js/blob/master/packages/client/src/index.ts#L223
```js
  /**
   * Start the sign-in flow with the specified redirect URI. The URI must be
   * registered in the Logto Console.
   *
   * The user will be redirected to that URI after the sign-in flow is completed,
   * and the client will be able to get the authorization code from the URI.
   * To fetch the tokens from the authorization code, use {@link handleSignInCallback}
   * after the user is redirected in the callback URI.
   *
   * @param redirectUri The redirect URI that the user will be redirected to after the sign-in flow is completed.
   * @param interactionMode The interaction mode to be used for the authorization request. Note it's not
   * a part of the OIDC standard, but a Logto-specific extension. Defaults to `signIn`.
   *
   * @see {@link https://docs.logto.io/docs/recipes/integrate-logto/vanilla-js/#sign-in | Sign in} for more information.
   * @see {@link InteractionMode}
   */
  async signIn(redirectUri: string, interactionMode?: InteractionMode): Promise<void> {
```
